### PR TITLE
upgrade to wgpu 22.0.0

### PR DESCRIPTION
**Pull request by KeKsBoTer**
wgpu changelog: https://github.com/gfx-rs/wgpu/releases/tag/v22.0.0

### CODE REVIEW

File: crates/egui-wgpu/src/renderer.rs [hunk 1]
```diff
@@ -345,6 +345,7 @@ impl Renderer {
                     compilation_options: wgpu::PipelineCompilationOptions::default()
                 }),
                 multiview: None,
+                cache:None,
```
Comment (torokati44) [inline crates/egui-wgpu/src/renderer.rs:348]: Huh, how come the format checker doesn't complain about the lack of space here...? 🤔

Comment (Wumpf) [reply crates/egui-wgpu/src/renderer.rs]: strange indeed, I poked it to format that file directly and it didn't care. hum.

**Comment by Wumpf** [comment]
Now that lifetimes on passes are more relaxed there's probably room for some follow-ups we can do to make the render callback system easier to use. Needs looking into & experimentation!
(meant to do that before wgpu 22 goes out as an exploration of wgpu's changes there, but things rarely go as planned :))

**Review by Wumpf — APPROVED** [review]

**Comment by John-Nagle** [comment]
It looks like wgpu-egui was changed without changing its version:

https://github.com/emilk/egui/commit/56df31ab489312014d57247db438ecd7189a710a

So that can't be pushed to crates.io, where the version must change if the content does.
So [egui-wgpu at crates.io](https://crates.io/crates/egui-wgpu) 0.28.1, last updated 14 days ago, is out of sync with Github master. I think.

**Comment by Wumpf** [comment]
nothing unusual there: egui & eframe also has changed since 0.28.1. There's a few repos that immediately change the version after release because _technically_ the first change after a release if for an unreleased version, but this repo is not one of them.
There's going to be a 0.29.0 at some point published to crates.io containing that change, until then applications that want it will have to point to a commit hash instead

**Comment by John-Nagle** [comment]
OK. So there's a workaround on the egui side.

I'm currently stuck due to wgpu-profiler problems. [That can't even be built against current WGPU without another WGPU fix.](https://github.com/Wumpf/wgpu-profiler/pull/75) Looks like a WGPU 22.1.x will be needed.

**Pull request merged**